src/App.tsx

```diff
@@ -1,24 +1,149 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Notification } from './components/Notification/Notification';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { TodoErrors } from './types/TodoErrors';
+import {
+  getTodos, USER_ID, createTodo, deleteTodo,
+} from './api/todos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<TodoErrors | null>(null);
+  const [status, setStatus] = useState<TodoStatus>(TodoStatus.All);
+  const [title, setTitle] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isProcessed, setIsProcessed] = useState<boolean>(false);
+  const [deletedId, setDeletedId] = useState<number | null>(null);
+
+  const deleteTodoHandler = (todoId: number) => {
+    deleteTodo(todoId)
+      .then(() => setDeletedId(todoId))
+      .catch(() => setError(TodoErrors.Delete));
+  };
+
+  const createTodoHandler = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      if (!title.trim()) {
+        setError(TodoErrors.Empty);
+      } else {
+        setTempTodo({
+          id: 0,
+          title,
+          completed: false,
+          userId: USER_ID,
+        });
+
+        createTodo(USER_ID, title)
+          .then(() => setTempTodo(null))
+          .catch(() => {
+            setError(TodoErrors.Add);
+          })
+          .finally(() => {
+            setIsProcessed(false);
+            setTitle('');
+          });
+        setIsProcessed(true);
+      }
+    }
+  };
+
+  const clearCompletedTodos = () => {
+    const completeTodos = todos.filter(todo => todo.completed);
+
+    return completeTodos
+      .map(todo => deleteTodo(todo.id)
+        .catch(() => setError(TodoErrors.Delete)));
+  };
+
+  const itemsLeft = todos.filter(todo => !todo.completed).length;
+  const itemsCompleted = todos.filter(todo => todo.completed).length;
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((fetchedTodos: Todo[]) => {
+        setTodos(fetchedTodos);
+      })
+      .catch(() => {
+        setError(TodoErrors.Get);
+      });
+  });
```

Comment:
you should add `[]` to avoid  this 
<img width="766" alt="image" src="https://user-images.githubusercontent.com/70220667/236665370-5554dcff-2e68-4c15-af4e-1eddede7ecbd.png">